index.js

```diff
@@ -52,322 +38,162 @@ class Client {
+            .then(() => {
+              this.serverless.cli.log(`Removing bucket...`);
+              return bucketUtils.deleteBucket(this.aws, bucketName);
+            })
+            .then(() => {
+              this.serverless.cli.log(
+                `Success! Your files have been removed and your bucket has been deleted`
+              );
+            });
+        } else {
+          this.serverless.cli.log(`Bucket does not exist`);
         }
-      };
-
-      return this.aws.request('S3', 'putBucketCors', params);
-    }
-
-    return this.listBuckets()
-      .then(this.findBucket)
-      .then(this.listObjectsInBucket)
-      .then(this.deleteObjectsFromBucket)
-      .then(createBucket)
-      .then(configureBucket)
-      .then(configurePolicyForBucket)
-      .then(configureCorsForBucket)
-      .then(function() {
-        return this._uploadDirectory(this.clientPath);
+      })
+      .catch(error => {
+        return BbPromise.reject(new this.error(error));
       });
   }
 
-  _uploadDirectory(directoryPath) {
-    let _this = this,
-      readDirectory = _.partial(fs.readdir, directoryPath);
-
-    async.waterfall([
-      readDirectory,
-      function(files) {
-        files = _.map(files, function(file) {
-          return path.join(directoryPath, file);
-        });
-
-        async.each(files, function(path) {
-          fs.stat(
-            path,
-            _.bind(function(err, stats) {
-              return stats.isDirectory() ? _this._uploadDirectory(path) : _this._uploadFile(path);
-            }, _this)
-          );
-        });
-      }
-    ]);
-  }
-
-  _uploadFile(filePath) {
-    let _this = this,
-      urlRoot = regionToUrlRootMap(_this.region);
-
-    let fileKey = path.normalize(filePath).replace(_this.clientPath, '');
-    if (fileKey.substr(0, 1) === path.sep) {
-      fileKey = fileKey.replace(path.sep, '');
-    }
-
-    let distRoot = path.join(
-      _this.serverless.config.servicePath,
-      _this.serverless.service.custom.client.distributionFolder || path.join('client', 'dist')
-    );
-    distRoot += path.sep;
-
-    let baseHeaderKeys = [
-      'Cache-Control',
-      'Content-Disposition',
-      'Content-Encoding',
-      'Content-Language',
-      'Content-Type',
-      'Expires',
-      'Website-Redirect-Location'
-    ];
+  _processDeployment() {
+    this._validateConfig();
 
-    let ruleList = [];
-    let headers = _this.serverless.service.custom.client.objectHeaders;
-    if (headers) {
-      if (headers.ALL_OBJECTS) {
-        ruleList = ruleList.concat(headers.ALL_OBJECTS);
-      }
-      Object.keys(headers)
-        .filter(m => m.substr(-1, 1) === '/') // folders
-        .sort((a, b) => a.length > b.length) // sort by length ascending
-        .forEach(m => {
-          if (filePath.replace(distRoot, '').substr(0, m.length) === m) {
-            ruleList = ruleList.concat(headers[m]);
+    const region =
+      this.cliOptions.region ||
```

Comment:
Can you add a comment on order of precedence here? Just to explain what each of these are pulling from: 

`CLI arguments override the config file which overrides the. . .`